### DISCUSSION TITLE
HELP!! Remote button just disappeared.

### DISCUSSION THREAD

**Discussion by frankdadale** [microsoft/vscode-jupyter#12999] [Questions and Answers] [answered]
**HELP!! The Remote button just disappeared.**
There should have a remote button on the right-bottom of my Jupyter workspace, but it's gone now. The kernel selection button on the right-top, not works, cannot add existed jupyter server.
What should I do to get my remote button back?

**Comment by DonJayamanne** [comment, marked as answer]
You can still add them, the UI has been streamlined, so that the place you add the Uri for the server is in the kernel picker itself
see here https://code.visualstudio.com/docs/datascience/jupyter-kernel-management#_existing-jupyter-server

Please let me know if you still run into any issues.

**Reply by frankdadale** [reply]
Not work.
When I add a jupyter server, error came out "Connection failure. Verify the server is running and reachable from a browser. (Forbidden). When 
connecting from vscode.dev Jupyter servers must be started with specific options to connect. Click
[here](https://github.com/microsoft/vscode-jupyter/wiki/Connecting-to-a-remote-Jupyter-server-from-vscode.dev) for more information. 
"
What's "connecting from vscode.dev Jupyter servers must be started with specific options to connect." means, No usefull answers are found in the link.

**Reply by DonJayamanne** [reply]
@frankdadale 
The error message is basically saying that the URL you have entered does not seem to be valid (or the fact that the jupyter extension is not able to verify the URL by connecting to it)

Is this a local Jupyter server? 
Is it behing a username/password
Does the URL contain the token information?

**Reply by frankdadale** [reply]
It's my local server with password xxx
URL: http://domain:8888
I don't have the token, before, through the remote server button, I could first type the URL, then type the password. This couldn't work now through the select kernel button.
By the way, I could access it through web browser.

**Reply by DonJayamanne** [reply]
Looks like you are running into this bug https://github.com/microsoft/vscode-jupyter/issues/13000#event-8677137032

**Reply by DonJayamanne** [reply]
See this for a temporary work around https://github.com/microsoft/vscode-jupyter/issues/13000#issuecomment-1457168208

**Reply by frankdadale** [reply]
Thanks very much!!!!
The remote button just come back. I could type the password now.

**Reply by DonJayamanne** [reply]
@frankdadale please could you confirm the fact that you need to type in the password everytime you re-load VS Code?

**Reply by frankdadale** [reply]
@DonJayamanne Yes, every time I re-open this remote workspace, I need to type the ssh password to open the remote files and jupyter-server password to connect to the kernel.
